keyvi/include/keyvi/vector/vector_generator.h

```diff
@@ -0,0 +1,91 @@
+/* * keyvi - A key value store.
+ *
+ * Copyright 2018   Narek Gharibyan<narekgharibyan@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ *  vector_generator.h
+ *
+ *  Created on: March 17, 2018
+ *      Author: Narek Gharibyan <narekgharibyan@gmail.com>
+ */
+
+#ifndef KEYVI_VECTOR_VECTOR_GENERATOR_H_
+#define KEYVI_VECTOR_VECTOR_GENERATOR_H_
+
+#include <string>
+
+#include <boost/filesystem/path.hpp>
+
+#include "dictionary/fsa/internal/constants.h"
+#include "util/configuration.h"
+#include "vector/vector_file.h"
+
+namespace keyvi {
+namespace vector {
+
+template <class ValueStoreT>
+class VectorGenerator final {
+  static const size_t INDEX_EXTERNAL_MEMORY_CHUNK_SIZE = sizeof(offset_type) * 100 * 1000 * 1000;
+
+  using parameters_t = keyvi::util::parameters_t;
+
+ public:
+  explicit VectorGenerator(const parameters_t& params_arg = parameters_t()) {
+    parameters_t params = params_arg;
+    params[TEMPORARY_PATH_KEY] = keyvi::util::mapGetTemporaryPath(params);
+    params[MINIMIZATION_KEY] = "off";
+
+    temporary_directory_ = params[TEMPORARY_PATH_KEY];
+    temporary_directory_ /= boost::filesystem::unique_path("keyvi-vector-%%%%-%%%%-%%%%-%%%%");
+    boost::filesystem::create_directory(temporary_directory_);
+
+    index_store_.reset(new MemoryMapManager(INDEX_EXTERNAL_MEMORY_CHUNK_SIZE, temporary_directory_, "index-chunk"));
+    value_store_.reset(new ValueStoreT(params));
+  }
+
+  VectorGenerator& operator=(VectorGenerator const&) = delete;
+
+  VectorGenerator(const VectorGenerator&) = delete;
+
+  ~VectorGenerator() { boost::filesystem::remove_all(temporary_directory_); }
+
+  void PushBack(typename ValueStoreT::value_t value) {
+    bool dummy_minimization = false;
+    const offset_type value_idx = value_store_->GetValue(value, &dummy_minimization);
+
+    index_store_->Append(&value_idx, sizeof(offset_type));
```

Comment:
it would be good to make this platform independent. This is easily done by defining your preferred endianess, I would go for little endian, which is Intel's default. 

You need to:

include endian.h

call htole64(v) for host to persistence order and le64toh(v) for persistence to host order on lookup. On Intel the macros are optimized out.